src/pageEditor/slices/editorSlice.ts

```
@@ -441,6 +448,47 @@ export const editorSlice = createSlice({
         element.recipe = metadata;
       }
     },
+    showAddToRecipeModal(state) {
+      console.log("show action");
+      state.isAddToRecipeModalVisible = true;
+    },
+    hideAddToRecipeModal(state) {
+      state.isAddToRecipeModalVisible = false;
+    },
+    addElementToRecipe(
+      state,
+      action: PayloadAction<{
+        elementId: UUID;
+        recipeMetadata: RecipeMetadata;
+        keepLocalCopy: boolean;
+      }>
+    ) {
+      const {
+        payload: { elementId, recipeMetadata, keepLocalCopy },
+      } = action;
+      const element = state.elements.find(
+        (element) => element.uuid === elementId
+      );
+      if (!element) {
```

Let's throw an error here? Prefer to throw errors instead of silently doing bad behavior